### PR TITLE
Correct `JavaGenericPlaceholderElement` native type

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy [hunk 1]
```diff
@@ -423,8 +423,14 @@ class Test<T> {
 }
 """)
         def fieldType = classElement.fields[0].type
+        def placeholders = fieldType.getDeclaredGenericPlaceholders()
 
         expect:
+        if (placeholders.size() > 0) {
```
Comment (sdelamo) [inline inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy]: you can rewrite this as: 

```groovy
            placeholders.each {
                 assert it.nativeType.class.simpleName == "TypeVar"
             }
```

to 

```groovy
assert placeholders.every { it.nativeType.class.simpleName == "TypeVar" }
```

File: inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy [hunk 2]
```diff
@@ -423,8 +423,14 @@ class Test<T> {
 }
 """)
         def fieldType = classElement.fields[0].type
+        def placeholders = fieldType.getDeclaredGenericPlaceholders()
```
Comment (sdelamo) [inline inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy]: Use type instead of `def`

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by graemerocher — APPROVED** [review]

File: inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy [hunk 3]
```diff
@@ -423,8 +423,10 @@ class Test<T> {
 }
 """)
         def fieldType = classElement.fields[0].type
+        List<? extends GenericPlaceholderElement> placeholders = fieldType.getDeclaredGenericPlaceholders()
 
         expect:
+        assert placeholders.every { it.nativeType.class.simpleName == "TypeVar" }
```
Comment (sdelamo) [inline inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy]: assert is not necessary here

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy (hunk 3 above)
Comment (sdelamo) [reply inject-java/src/test/groovy/io/micronaut/annotation/processing/visitor/JavaReconstructionSpec.groovy]: resolved via https://github.com/micronaut-projects/micronaut-core/pull/7081/commits/34c232a5b4bc9f909aaa37fde496e4bf10a68c63

**Review by sdelamo — APPROVED** [review]

**Pull request merged**